From 6b8d4ba966a5c3f8b210d394cd74e3b037e8266e Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@nabladev.com>
Date: Wed, 8 Oct 2025 13:41:02 +0200
Subject: [PATCH 9000/9001] FIXME: arm64: dts: st: Switch flexgen27 clock to
 syscfg clock hack

The clocking of display pipeline on STM32MP25xx is horribly misdesigned
and full of workarounds for broken SCMI clock and other workarounds for
broken clock tree. Switch the lcd clock to one of the workaround syscon
clock so the LTDC can be clocked from its kernel clock, which is needed
for DSI to work.

Signed-off-by: Marek Vasut <marex@nabladev.com>
---
 arch/arm64/boot/dts/st/stm32mp255.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp255.dtsi b/arch/arm64/boot/dts/st/stm32mp255.dtsi
index a98b59f47434d..5cb56d4f3ad76 100644
--- a/arch/arm64/boot/dts/st/stm32mp255.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp255.dtsi
@@ -29,7 +29,7 @@ map1 {
 
 &ltdc {
 	compatible = "st,stm32mp255-ltdc";
-	clocks = <&clk_flexgen_27_fixed>, <&rcc CK_BUS_LTDC>, <&syscfg>, <&lvds>;
+	clocks = <&syscfg>, <&rcc CK_BUS_LTDC>, <&rcc CK_KER_LTDC>, <&lvds>;
 	clock-names = "lcd", "bus", "ref", "lvds";
 };
 
-- 
2.53.0

